src/s2/base/port.h

```diff
@@ -602,11 +602,11 @@ std::ostream &operator<<(std::ostream &out, const pthread_t &thread_id);
 #endif  // _MSC_VER
 
 // random, srandom
-#ifdef _MSC_VER
+#if defined(_MSC_VER) || defined(__MINGW32__)
```

Comment:
Everything that's actually related to windows rather than MSVC should be switched to `_MSC_VER`.